cluster/local/integration_tests_tls.sh

```diff
@@ -0,0 +1,354 @@
+metadata:
+  name: debug-config
+spec:
+  args:
+  - --debug
+---
+apiVersion: pkg.crossplane.io/v1
+kind: Provider
+metadata:
+  name: "${PACKAGE_NAME}"
+spec:
+  controllerConfigRef:
+    name: debug-config
+  package: "${PACKAGE_NAME}"
+  packagePullPolicy: Never
+EOF
+)"
+
+echo "${INSTALL_YAML}" | "${KUBECTL}" apply -f -
+
+# printing the cache dir contents can be useful for troubleshooting failures
+echo_step "check kind node cache dir contents"
+docker exec "${K8S_CLUSTER}-control-plane" ls -la /cache
+
+echo_step "waiting for provider to be installed"
+"${KUBECTL}" wait "provider.pkg.crossplane.io/${PACKAGE_NAME}" --for=condition=healthy --timeout=60s
+
+# Generate CA key and certificate
+openssl genrsa -out ca-key.pem 2048
+openssl req -new -x509 -key ca-key.pem -out ca-cert.pem -days 365 -subj "/CN=CA"
+
+# Generate server key and certificate
+openssl genrsa -out server-key.pem 2048
+openssl req -new -key server-key.pem -out server-req.pem -subj "/CN=mariadb.default.svc.cluster.local"
+openssl x509 -req -in server-req.pem -CA ca-cert.pem -CAkey ca-key.pem -CAcreateserial -out server-cert.pem -days 365
+
+# Generate client key and certificate
+openssl genrsa -out client-key.pem 2048
+openssl req -new -key client-key.pem -out client-req.pem -subj "/CN=client"
+openssl x509 -req -in client-req.pem -CA ca-cert.pem -CAkey ca-key.pem -CAcreateserial -out client-cert.pem -days 365
+
+# Create a secret for the server TLS certificates and keys
+#"${KUBECTL}" delete secret mariadb-server-tls || true
+"${KUBECTL}" create secret generic mariadb-server-tls \
+    --from-file=ca-cert.pem \
+    --from-file=server-cert.pem \
+    --from-file=server-key.pem
+
+# Create a secret for the client TLS certificates and keys
+#"${KUBECTL}" delete secret mariadb-client-tls || true
+"${KUBECTL}" create secret generic mariadb-client-tls \
+    --from-file=ca-cert.pem \
+    --from-file=client-cert.pem \
+    --from-file=client-key.pem
+
+# install MariaDB chart with TLS enabled
+echo_step "installing MariaDB Helm chart with TLS into default namespace"
+mariadb_root_pw=$(openssl rand -base64 32)
+mariadb_test_pw=$(openssl rand -base64 32)
+
+"${HELM3}" repo add bitnami https://charts.bitnami.com/bitnami
+
+MARIADB_VALUES=$(cat <<EOF
+auth:
+  rootPassword: ${mariadb_root_pw}
+primary:
+  extraFlags: "--ssl --require-secure-transport=ON --ssl-ca=/opt/bitnami/mariadb/certs/ca-cert.pem --ssl-cert=/opt/bitnami/mariadb/certs/server-cert.pem --ssl-key=/opt/bitnami/mariadb/certs/server-key.pem"
+  configurationSecret: mariadb-server-tls
+  extraVolumes:
+    - name: tls-certificates
+      secret:
+        secretName: mariadb-server-tls
+  extraVolumeMounts:
+    - name: tls-certificates
+      mountPath: /opt/bitnami/mariadb/certs
+      readOnly: true
+initdbScripts:
+  init.sql: |
+    CREATE USER 'test'@'%' IDENTIFIED BY '${mariadb_test_pw}' REQUIRE X509;
+    GRANT ALL PRIVILEGES ON *.* TO 'test'@'%' WITH GRANT OPTION;
+    FLUSH PRIVILEGES;
```

Comment:
A dedicated `test` user is required in order to specifically require X509 on it, but not for `admin` user, which is also used for health probes without TLS.